core-overlay.html

```diff
@@ -111,666 +124,373 @@
+        }
+      },
 
-    discoverDimensions: function() {
-      if (this.dimensions) {
-        return;
-      }
-      var target = getComputedStyle(this.target);
-      var sizer = getComputedStyle(this.sizingTarget);
-      this.dimensions = {
-        position: {
-          v: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
-            'bottom' : null),
-          h: target.left !== 'auto' ? 'left' : (target.right !== 'auto' ?
-            'right' : null),
-          css: target.position
-        },
-        size: {
-          v: sizer.maxHeight !== 'none',
-          h: sizer.maxWidth !== 'none'
-        },
-        margin: {
-          top: parseInt(target.marginTop) || 0,
-          right: parseInt(target.marginRight) || 0,
-          bottom: parseInt(target.marginBottom) || 0,
-          left: parseInt(target.marginLeft) || 0
+      renderBackdropOpened: function() {
+        if (this.backdrop && manager.getBackdrops().length < 2) {
+          this._backdrop.classList.toggle('core-opened', this.opened);
         }
-      };
-    },
-
-    finishPositioning: function(target) {
-      this.target.style.display = 'none';
-      this.target.style.transform = this.target.style.webkitTransform = '';
-      // force layout to avoid application of transform
-      this.target.offsetWidth;
-      this.target.style.transition = this.target.style.webkitTransition = '';
-    },
-
-    getTransition: function(name) {
-      return this.meta.byId(name || this.transition);
-    },
-
-    getFocusNode: function() {
-      return this.target.querySelector('[autofocus]') || this.target;
-    },
-
-    applyFocus: function() {
-      var focusNode = this.getFocusNode();
-      if (this.opened) {
-        if (!this.autoFocusDisabled) {
-          focusNode.focus();
+      },
+
+      completeBackdrop: function() {
+        if (this.backdrop) {
+          manager.trackBackdrop(this);
+          if (manager.getBackdrops().length === 0 && this._backdrop.parentNode) {
+            Polymer.dom(this._backdrop.parentNode).removeChild(this._backdrop);
+          }
         }
-      } else {
-        focusNode.blur();
-        if (currentOverlay() == this) {
-          console.warn('Current core-overlay is attempting to focus itself as next! (bug)');
+      },
+
+      preparePositioning: function() {
+        this.style.transition = this.style.webkitTransition = 'none';
+        this.style.transform = this.style.webkitTransform = 'none';
+        this.style.display = '';
+      },
+
+      finishPositioning: function(target) {
+        this.style.display = 'none';
+        this.style.transform = this.style.webkitTransform = '';
+        // force layout to avoid application of transform
+        this.offsetWidth;
+        this.style.transition = this.style.webkitTransition = '';
+      },
+
+      getFocusNode: function() {
+        return Polymer.dom(this).querySelector('[autofocus]') || this;
+      },
+
+      applyFocus: function() {
```

Comment:
It would be nice to integrate this behavior into `core-focused-state` if it is generally useful. What do you think?